### PR TITLE
Allow iterable dataset for detector fit

### LINKED ISSUE #15: Allow iterable dataset for detector fit
Right now it's possible to fit the detectors only on map-style datasets, as https://github.com/TorchDrift/TorchDrift/blob/master/torchdrift/utils/fit.py#L36 calls the `len()` of the dataset/dataloader, which is not guaranteed to be implemented on an iterable dataset.

It could be possible to refactor the method to allowed both dataset types.

### PR DESCRIPTION
**Pull request by alessiamarcolini**
Fixes #15

### CODE REVIEW

File: torchdrift/utils/fit.py [hunk 1]
```diff
@@ -33,14 +33,24 @@ def fit(
...
+    if hasattr(dl.dataset, "__len__"):
+        nb = len(dl)
+        if num_batches is not None:
+            nb = min(nb, num_batches)
+        total = nb
+    else:
+        total = None
+
+    for i, b in enumerate(tqdm.tqdm(dl, total=total)):
+        if num_batches and i >= num_batches:
```
Comment (lantiga) [inline torchdrift/utils/fit.py:46]:
```suggestion
        if num_batches is not None and i >= num_batches:
```

**Review by lantiga — CHANGES_REQUESTED** [review]
Thank you @alessiamarcolini! 
Looks great, only a minor nit

**Review by lantiga — APPROVED** [review]

**Comment by alessiamarcolini** [comment]
Hi @lantiga, I don't think the failures are related to my changes :D

**Comment by t-vi** [comment]
Looks awesome, thank you @alessiamarcolini .

**Pull request merged**